src/history.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * This module makes the different routes created with react-chicane listen to the same history instance
+ */
+import { createBrowserHistory, createPath, parsePath } from "history";
+import * as React from "react";
+import { useSubscription } from "use-subscription";
+import { areLocationsEqual, decodeLocation } from "./location";
+import { Location, Subscription } from "./types";
+const subscriptions = new Set<Subscription>();
+
+export const history = createBrowserHistory();
+
+let currentLocation = decodeLocation(history.location, true);
+
+export const getCurrentLocation = () => {
+  return currentLocation;
+};
+
+if (currentLocation.url !== createPath(history.location)) {
+  history.replace(currentLocation.url); // URL cleanup
+}
+
+history.listen(({ location }) => {
+  const nextLocation = decodeLocation(location, false);
+  if (!areLocationsEqual(nextLocation, currentLocation)) {
```

Comment:
The location update event need to be emitted.
Let's suppose you have a link to the current page, not disabled with some `active` link check. You click on it. The focus need be reset to the content (in my opinion).

Reply:
Hum, in the current state that would not happen even if we fired that event, as the route itself wouldn't change

Reply:
In the current state `useLocation()` and `subscribe` would be triggered, but indeed not `useRoute`. Ok for me then.